File: backend/src/main/java/com/farmersbuddy/farmers_buddy_backend/service/ToolService.java
package com.farmersbuddy.farmers_buddy_backend.service;

import com.farmersbuddy.farmers_buddy_backend.entity.Tool;
import com.farmersbuddy.farmers_buddy_backend.repository.ToolRepository;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

// =============================================================
// ToolService.java — Business Logic for Tools Catalog Module
// =============================================================
// Handles adding, fetching, and deleting tools from the catalog.
// Sits between ToolController and ToolRepository.
//
// Architecture: ToolController → ToolService → ToolRepository → MySQL
// =============================================================

@Service
public class ToolService {

    @Autowired
    private ToolRepository toolRepository;

    // Save a new tool to the catalog (admin action)
    public Tool addTool(Tool tool) {
        return toolRepository.save(tool);
    }

    // Return all tools in the catalog (farmer view)
    public List<Tool> getAllTools() {
        return toolRepository.findAll();
    }

    // Remove a tool by ID (admin action)
    public void deleteTool(Long id) {
        toolRepository.deleteById(id);
    }
}

File: frontend/src/services/toolService.js
// =============================================================
// toolService.js — Tools Catalog API Calls
// =============================================================
// Handles all HTTP calls for the dynamic Tools Catalog module.
// Admin can add/delete tools; all users can browse.
// =============================================================

import API from "./api"

// Admin adds a new farming tool to the catalog
export const addTool = (data) =>
  API.post("/tools", data).then((res) => res.data)

// All users browse the full tools catalog
export const getAllTools = () =>
  API.get("/tools").then((res) => res.data)

// Admin removes a tool from the catalog by ID
export const deleteTool = (id) =>
  API.delete(`/tools/${id}`).then((res) => res.data)
